src/window_internal_glutin.rs

```diff
@@ -522,6 +565,21 @@ impl<UserEventType: 'static> WindowGlutin<UserEventType>
                     handler.on_keyboard_modifiers_changed(helper, state.into())
                 }
 
+                GlutinWindowEvent::HoveredFile(path) => {
+                    helper.set_drag_file(path, DragFileState::Hover);
```

Comment:
I think it would be simpler to avoid storing anything in the `WindowHelper`, and just have one callback `on_file_drag()`, which has an enum as a parameter:

```rust
pub enum FileDragState
{
    Hovered(PathBuf),
    Dropped(PathBuf),
    Cancelled
}
```